service/block/event.go

```diff
@@ -60,12 +60,22 @@ func (s *Service) handleRawBlock(raw *RawBlock) error {
 		return err
 	}
 	log.Debugw("generated DataAvailabilityHeader", "data root", dah.Hash())
+	// create ExtendedHeader
+	commit, err := s.fetcher.CommitAtHeight(context.Background(), &raw.Height)
+	if err != nil {
+		log.Errorw("fetching commit", "err", err.Error(), "height", raw.Height)
+		return err
+	}
+	extendedHeader := &header.ExtendedHeader{
+		RawHeader: raw.Header,
+		DAH:       &dah,
+		Commit:    commit,
+		// TODO @renaynay: validator set
```

Comment:
Either create an issue or implement it in this PR. It is a no-brainer to find another RPC call to get ValidatorSet and fill the Header here fully.

Reply:
Wait, the extended header contains _the entire_ active (?) validator set?

Reply:
Isn't this a requirement for validation purposes? Commit verification requires the whole validator set.

Reply:
Yes, the validator set is required to validate a Commit. But I had assumed the node would keep track of the validator set as it changed each block. Does it validate the Commit statelessly?

Reply:
We will have ValidatorSet in header sync logic state; the question is how it gets updated? My current understanding is that after newly received ExtendedHeader is verified by the existing ValidatorSet in the state, a new ValidatorSet in the new header is selected instead.